userspace/libsinsp/test/events_fspath.ut.cpp

```diff
@@ -104,6 +107,13 @@ class fspath : public sinsp_with_test_input
 		ASSERT_FALSE(field_has_value(evt, fs_path_targetraw));
 	}
 
+	void verify_fd_name_same_fs_path_name(sinsp_evt *evt)
+	{
+		// todo the cached fd.name might be subject to some std::string_view lifetime issues as the second path of the path concatenation
+		// is missing when retrieving fd.name from the cache. Let's check if string_view and std::string handling is done correctly throughout.
+		// ASSERT_STREQ(get_field_as_string(evt, fs_path_name).c_str(), get_field_as_string(evt, "fd.name").c_str());
```

Comment:
@falcosecurity/libs-maintainers and also @federico-sysdig would appreciate some help to resolve some mystery (again) ....

So these tests (comparing the `fd.name` from the cache to the `fs.path.name`) are failing, possibly due to some std::string_view lifetime issues or other issues? Regardless am stuck debugging.

Sprinkled few print statements along the relevant code paths and we only loose the second part of the concatenated path when retrieving the name from the cache in the fd filterchecks. Before that it's all correct within the parsers logic.

```
[ RUN      ] fspath.openat_2_relative
parser name exit tmp/random/dir...///../../name.txt
parser name enter event tmp/random/dir...///../../name.txt
parser name fullpath /tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8/tmp/name.txt
parser name fullpath from fdi /tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8/tmp/name.txt
parser name fullpath from table cache fdi /tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8/tmp/name.txt
FDNAME fd 3 CACHE FILTERCHECK /tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8
```

[Btw the fallback in the filtercheck `extract_from_null_fd` is working as expected and these tests would pass.  Also note that today we do not test `fd.name` for many scenarios ...]

Reply:
I don't understand. Are you saying that you get a full path when using `get_field_as_string(evt, fs_path_name)` and only the first part of it when using `get_field_as_string(evt, "fd.name")`? Should they really be equal? Wouldn't it depend on the value of `fs_path_name`?
If there is an issue with a string view pointing at a string whose lifetime is over, that is indeed a serious issue, but it would be a problem within the machinery behind `get_field_as_string`.

Nit: The check can be a bit simpler:
```c++
ASSERT_EQ(get_field_as_string(evt, fs_path_name), get_field_as_string(evt, "fd.name"));
```

Reply:
(1) Incorporated your nit suggestion  and also made sure the list of events where these 2 should be same is complete. Re-pushed the last commit.

(2) Plus let me share the 2 locations in the source code where the cached string that represents the fd full file path is not equal. 

- https://github.com/falcosecurity/libs/blob/874e573b20178325ebe483af7f104ae3c70fdf34/userspace/libsinsp/sinsp_filtercheck_fd.cpp#L520 which is `m_tstr = m_fdinfo->m_name;` has the second part of the paths that were concatenated missing
- Versus in the parsers https://github.com/falcosecurity/libs/blob/874e573b20178325ebe483af7f104ae3c70fdf34/userspace/libsinsp/parsers.cpp#L2776 `m_name`  is correct, also when directly retrieving it again after having written it to the fd table / cache.

See below a print output for how https://github.com/falcosecurity/libs/blob/874e573b20178325ebe483af7f104ae3c70fdf34/userspace/libsinsp/sinsp_filtercheck_fd.cpp#L520 `m_tstr = m_fdinfo->m_name;` looks like atm.

```
[ RUN      ] fspath.open

FILTERCHECK FDNAME FROM FD CACHE /tmp/name

[       OK ] fspath.open (0 ms)
[ RUN      ] fspath.openat
[       OK ] fspath.openat (0 ms)
[ RUN      ] fspath.openat_2
FILTERCHECK FDNAME FROM FD CACHE /tmp/name
[       OK ] fspath.openat_2 (0 ms)
[ RUN      ] fspath.openat_2_relative

FILTERCHECK FDNAME FROM FD CACHE /tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8

/home/m/Documents/OSS/oss-dev-master/libs/userspace/libsinsp/test/events_fspath.ut.cpp:114: Failure
Expected equality of these values:
  get_field_as_string(evt, fs_path_name)
    Which is: "/tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8/tmp/name.txt"
  get_field_as_string(evt, "fd.name")
    Which is: "/tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8"
[  FAILED  ] fspath.openat_2_relative (0 ms)
[ RUN      ] fspath.openat2

FILTERCHECK FDNAME FROM FD CACHE /tmp/name

[       OK ] fspath.openat2 (0 ms)
[ RUN      ] fspath.openat2_relative

FILTERCHECK FDNAME FROM FD CACHE /tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8

/home/m/Documents/OSS/oss-dev-master/libs/userspace/libsinsp/test/events_fspath.ut.cpp:114: Failure
Expected equality of these values:
  get_field_as_string(evt, fs_path_name)
    Which is: "/tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8/tmp/name.txt"
  get_field_as_string(evt, "fd.name")
    Which is: "/tmp/dirfd1/dirfd2/dirfd3/dirfd4/dirfd5/dirfd6/dirfd7/dirfd8"
[  FAILED  ] fspath.openat2_relative (0 ms)
[ RUN      ] fspath.openat2_relative_dirfd_cwd
```

Reply:
I'm taking a look at this. I tried pulling your branch, rebasing it and running the tests but this one doesn't seem to be failing. Did you happen to remove the failing one?

Reply:
The failing test is commented out for now.

Reply:
Thanks so much :rocket: @LucaGuerra. Pushed a co-authored commit. Feel free to overwrite it if you have a better idea.
Also created a follow ticket here: https://github.com/falcosecurity/libs/issues/2039

The macos test is currently failing ...